Sources/Sentry/SentryNSProcessInfoWrapper.h

```diff
@@ -0,0 +1,16 @@
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface SentryNSProcessInfoWrapper : NSObject
```

Comment:
`m`: This should be in `Sources/Sentry/include`. Same applies for the other non public header files in this PR. Some CI job should actually fail because of that.

Reply:
Looks like this would break SPM? We should put together some smoke tests for that/carthage/cocoapods integration. See https://github.com/getsentry/sentry-cocoa/issues/2517

Reply:
We already have CI actions to compile with SPM, CocoaPod and Carthage.

So I guess, if CI is not failing it means it is not breaking SPM. Every header inside "Sources/Sentry" is considered project header by SPM and Cocoapod. Carthage dont carry at all where the header is because depends of xcodeproj configuration.

We probably should create a new action only to check header locations.